src/test/ui/parser/lex-bad-char-literals-6.stderr

```diff
@@ -52,6 +53,7 @@ LL |     if x == z {}
    |          ^^ no implementation for `&str == char`
    |
    = help: the trait `std::cmp::PartialEq<char>` is not implemented for `&str`
+   = note: `std::cmp::PartialEq<char>` is implemented for `&mut str`, but not for `&str`
```

Comment:
These I believe are incorrect, I probably need to change the call to `predicate_may_hold` and instead check the resulting `EvaluationResult` from `evaluate_obligation_no_overflow`. Does that sound correct?